hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -1054,129 +1175,55 @@ int SaraNcpClient::selectSimCard(ModemState& state) {
         CHECK(waitAtResponseFromPowerOn(state));
     }
 
-    // Using numeric CME ERROR codes
-    // int r = CHECK_PARSER(parser_.execCommand("AT+CMEE=2"));
-    // CHECK_TRUE(r == AtResponse::OK, SYSTEM_ERROR_AT_NOT_OK);
+    // Checking for SIM readiness ensures that other related commands
+    // like IFC or PSM/eDRX won't error out
+    CHECK(checkSimReadiness(true));
 
-    int simState = 0;
-    unsigned attempts = 0;
-    for (attempts = 0; attempts < 10; attempts++) {
-        simState = checkSimCard();
-        if (!simState) {
-            break;
-        }
-        HAL_Delay_Milliseconds(1000);
-    }
+    return SYSTEM_ERROR_NONE;
+}
 
-    if (simState != SYSTEM_ERROR_NONE) {
-        return simState;
-    }
+int SaraNcpClient::disablePsmEdrx() {
 
-    if (attempts != 0 && ncpId() == PLATFORM_NCP_SARA_R410) {
-        // There was an error initializing the SIM
-        // This often leads to inability to talk over the data (PPP) muxed channel
-        // for some reason. Attempt to cycle the modem through minimal/full functional state.
-        // We only do this for R4-based devices, as U2-based modems seem to fail
-        // to change baudrate later on for some reason
-        CHECK_PARSER_OK(parser_.execCommand(UBLOX_CFUN_TIMEOUT, "AT+CFUN=0,0"));
-        CHECK_PARSER_OK(parser_.execCommand(UBLOX_CFUN_TIMEOUT, "AT+CFUN=1,0"));
+    // XXX: In theory, it's good to check for SIM readiness as PSM/eDRX commands seem
+    // to depend on it's readiness. However, if we have reached this point, it means
+    // that sim readiness is already checked
+    // CHECK(checkSimReadiness());
+
+    // Force Power Saving mode to be disabled
+    //
+    // TODO: if we enable this feature in the future add logic to CHECK_PARSER macro(s)
+    // to wait longer for device to become active (see MDMParser::_atOk)
+    auto resp = parser_.sendCommand("AT+CPSMS?");
+    unsigned psmState = 1;
+    auto r = resp.scanf("+CPSMS:%d", &psmState);
+    resp.readResult();
```

Comment:
`CHECK_PARSER(resp.readResult());`

We don't need to check for OK response, but generic errors should still be handled (e.g. a case where we've completely failed to get a response, or low level communication error).